build.gradle.kts

```diff
@@ -0,0 +1,234 @@
+        classDirectories = files(java.sourceSets["main"].output)
+    }
+}
+
+/*
+     * Allows you to run the UI tests in headless mode by calling gradle with the -Pheadless argument
+     */
+if (project.hasProperty("jenkinsBuild") || project.hasProperty("headless")) {
+    println("Running UI Tests Headless")
+    junitPlatform {
+        filters {
+            tags {
+                /*
+                 * A category for UI tests that cannot run in headless mode, ie work properly with real windows
+                 * but not with the virtualized ones in headless mode.
+                 */
+                exclude("NonHeadlessTests")
+            }
+        }
+    }
+    tasks {
+        "junitPlatformTest"(JavaExec::class) {
+            jvmArgs = listOf(
+                    "-Djava.awt.headless=true"
+            )
+        }
+    }
+}
+
+publishing {
+    publications {
+        create<MavenPublication>("RobotBuilder") {
+            artifactId = "RobotBuilder"
+            getWPILibVersion()?.let { version = it }
+            shadow.component(this)
+        }
+    }
+}
+
+// Ensure that the WPILibVersioningPlugin is setup by setting the release type, if releaseType wasn't
+// already specified on the command line
+if (!hasProperty("releaseType")) {
+    WPILibVersion {
+        releaseType = ReleaseType.DEV
+    }
+}
+
+/**
+ * @return [edu.wpi.first.wpilib.versioning.WPILibVersioningPluginExtension.version] value or null
+ * if that value is the empty string.
+ */
+fun getWPILibVersion(): String? = if (WPILibVersion.version != "") WPILibVersion.version else null
+
+task<Wrapper>("wrapper") {
+    gradleVersion = "4.1"
+}
+
+/**
+ * Retrieves the [java][org.gradle.api.plugins.JavaPluginConvention] project convention.
+ */
+val Project.`java`: org.gradle.api.plugins.JavaPluginConvention get() =
+    convention.getPluginByName("java")
+
+/**
+ * Retrieves the [checkstyle][org.gradle.api.plugins.quality.CheckstyleExtension] project extension.
+ */
+val Project.`checkstyle`: org.gradle.api.plugins.quality.CheckstyleExtension get() =
+    extensions.getByName("checkstyle") as org.gradle.api.plugins.quality.CheckstyleExtension
+
+/**
+ * Configures the [checkstyle][org.gradle.api.plugins.quality.CheckstyleExtension] project extension.
+ */
+fun Project.`checkstyle`(configure: org.gradle.api.plugins.quality.CheckstyleExtension.() -> Unit) =
+        extensions.configure("checkstyle", configure)
+
+/**
+ * Retrieves the [pmd][org.gradle.api.plugins.quality.PmdExtension] project extension.
+ */
+val Project.`pmd`: org.gradle.api.plugins.quality.PmdExtension get() =
+    extensions.getByName("pmd") as org.gradle.api.plugins.quality.PmdExtension
+
+/**
+ * Configures the [pmd][org.gradle.api.plugins.quality.PmdExtension] project extension.
+ */
+fun Project.`pmd`(configure: org.gradle.api.plugins.quality.PmdExtension.() -> Unit) =
+        extensions.configure("pmd", configure)
```

Comment:
Using the `plugins` syntax you shouldn't need a lot of this.

Reply:
this was copy and paste from OutlineViewer's script..